### PR TITLE
[fix-capture-org-changes] Update latest revision numbers when committing to GitHub

### PR DESCRIPTION
**Pull request by wlonk**
collie

### CODE REVIEW

File: metashare/api/sf_org_changes.py [hunk 1]
```diff
@@ -56,15 +56,34 @@ def commit_changes_to_github(
...
+                except KeyError:
+                    member_name_dict = scratch_org.latest_revision_numbers[
+                        member_name
+                    ] = {}
+
+                member_name_dict[member_type] = latest_revision_numbers[member_name][
+                    member_type
+                ]
+
+        scratch_org.save()
```
Comment (jgerigmeyer) [inline metashare/api/sf_org_changes.py]: Could we return the `latest_revision_numbers` here and do the update/save along with the other items back in the `job` (which also means after `refresh_from_db`)?

File: metashare/api/jobs.py [hunk 2]
```diff
@@ -179,9 +179,24 @@ def commit_changes_from_org(scratch_org, user, desired_changes, commit_message):
...
+                except KeyError:
+                    member_name_dict = scratch_org.latest_revision_numbers[
+                        member_name
+                    ] = {}
+
+                member_name_dict[member_type] = latest_revision_numbers[member_name][
+                    member_type
+                ]
+
+        scratch_org.save()
```
Comment (jgerigmeyer) [inline metashare/api/jobs.py]: We don't need this save; it'll happen in the except/else branches.

File: metashare/api/jobs.py [hunk 3]
```diff
@@ -195,6 +195,11 @@ def commit_changes_from_org(scratch_org, user, desired_changes, commit_message):
                 member_name_dict[member_type] = latest_revision_numbers[member_name][
                     member_type
                 ]
```
Comment (davisagli) [inline metashare/api/jobs.py]: This is building member_name_dict and then...doing nothing with it?

Comment (wlonk) [reply metashare/api/jobs.py]: We're mutating a dict that's on the `scratch_org` here and then saving it.

File: metashare/api/gh.py [hunk 4]
```diff
@@ -137,6 +139,7 @@ def extract_zip_file(zip_file, owner, repo_name):
     for path in itertools.chain(glob("zipball_root/*"), glob("zipball_root/.*")):
         shutil.move(path, ".")
     shutil.rmtree("zipball_root")
+    shutil.rmtree(ZIP_FILE_NAME)
```
Comment (jgerigmeyer) [inline metashare/api/gh.py]: Maybe should be `os.remove`?

```
HTTP POST /api/scratch-orgs/o1e03y8/commit/ 202 [0.05, 127.0.0.1:55980]
Traceback (most recent call last):
  File "/app/metashare/api/jobs.py", line 168, in commit_changes_from_org
    commit_message=commit_message,
  File "/app/metashare/api/sf_org_changes.py", line 58, in commit_changes_to_github
    with local_github_checkout(user, repo_url) as project_path:
  File "/usr/local/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/app/metashare/api/gh.py", line 161, in local_github_checkout
    extract_zip_file(zip_file, owner, repo_name)
  File "/app/metashare/api/gh.py", line 142, in extract_zip_file
    shutil.rmtree(ZIP_FILE_NAME)
  File "/usr/local/lib/python3.7/shutil.py", line 491, in rmtree
    _rmtree_safe_fd(fd, path, onerror)
  File "/usr/local/lib/python3.7/shutil.py", line 410, in _rmtree_safe_fd
    onerror(os.scandir, path, sys.exc_info())
  File "/usr/local/lib/python3.7/shutil.py", line 406, in _rmtree_safe_fd
    with os.scandir(topfd) as scandir_it:
NotADirectoryError: [Errno 20] Not a directory: 'archive.zip'
```

Comment (wlonk) [reply metashare/api/gh.py]: Should be; I thought that call was flexible. Will fix.

Comment (jgerigmeyer) [reply metashare/api/gh.py]: After making that change locally, now I get:

```
Traceback (most recent call last):
  File "/app/metashare/api/jobs.py", line 168, in commit_changes_from_org
    commit_message=commit_message,
  File "/app/metashare/api/sf_org_changes.py", line 63, in commit_changes_to_github
    user, scratch_org, os.path.join(project_path, "src"), desired_changes
  File "/app/metashare/api/sf_org_changes.py", line 43, in run_retrieve_task
    "commit": branch,
  File "/usr/local/lib/python3.7/site-packages/cumulusci/core/runtime.py", line 32, in __init__
    self._load_project_config(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/cumulusci/core/runtime.py", line 82, in _load_project_config
    self.global_config, *args, **kwargs
  File "/usr/local/lib/python3.7/site-packages/cumulusci/core/config/BaseProjectConfig.py", line 48, in __init__
    super(BaseProjectConfig, self).__init__(config=config)
  File "/usr/local/lib/python3.7/site-packages/cumulusci/core/config/BaseConfig.py", line 15, in __init__
    self._load_config()
  File "/usr/local/lib/python3.7/site-packages/cumulusci/core/config/BaseProjectConfig.py", line 75, in _load_config
    self.config_filename, repo_root
cumulusci.core.exceptions.ProjectConfigNotFound: The file cumulusci.yml was not found in the repo root: /tmp/tmp2zbduj9i/src. Are you in a CumulusCI Project directory?
```

**Pull request merged**